tests/unit/config_tests.py

```diff
@@ -236,6 +236,41 @@ def test_fallback_option_works(self):
         self.assertAlmostEqual(parser(".44"), .44)
 
 
+class DictOfTests(unittest.TestCase):
+    def test_empty(self):
+        parser = config.DictOf(config.Integer)
+        result = parser.parse("my_key", {"not_related": "a"})
+        self.assertEqual(result, {})
+
+    def test_scalar_children(self):
+        parser = config.DictOf(config.Integer)
+        result = parser.parse("my_key", {
+            "my_key.a": "1",
+            "my_key.b": "2",
+        })
+        self.assertEqual(result, {"a": 1, "b": 2})
+
+    def test_vector_children(self):
+        parser = config.DictOf({"a": config.Integer, "b": config.String})
+        result = parser.parse("my_key", {
+            "my_key.first.a": "1",
+            "my_key.first.b": "test",
```

Comment:
what happens if you add `"my_key.first.c"`?  I assume it just gets ignored?  It might be good to make that a part of the test.